source/docs/software/wpilib-tools/robotbuilder/writing-code/robotbuilder-drive-tank.rst

```diff
@@ -0,0 +1,266 @@
+.. image:: images/driving-with-joysticks-speed-controller-group.png
+
+Since we want to use more then two motors to drive the robot, inside the Differential Drive, create two Speed Controller Groups. These will group multiple speed controllers so they can be used with Differential Drive.
+
+.. image:: images/driving-with-joysticks-speed-controller.png
+
+Finally, create two Speed Controllers in each Speed Controller Group.
+
+Add the Joysticks to the Operator Interface
+-------------------------------------------
+
+.. image:: images/driving-with-joysticks-joysticks.png
+
+Add two joysticks to the Operator Interface, one is the left stick and the other is the right stick. The y-axis on the two joysticks are used to drive the robots left and right sides.
+
+.. note:: Be sure to export your program to C++ or Java before continuing to the next step.
+
+Create a Method to Write the Motors on the Subsystem
+----------------------------------------------------
+
+.. tabs::
+
+   .. group-tab:: java
+
+      .. code-block:: java
+         :emphasize-lines: 52-54
+
+         public class Drivetrain extends SubsystemBase {
+
+             // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+         private PWMVictorSPX left1;
+         private PWMVictorSPX left2;
+         private SpeedControllerGroup leftMotor;
+         private PWMVictorSPX right1;
+         private PWMVictorSPX right2;
+         private SpeedControllerGroup rightMotor;
+         private DifferentialDrive drive;
+
+             // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+
+             public Drivetrain() {
+                 // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+         left1 = new PWMVictorSPX(0);
+          addChild("left1",left1);
+          left1.setInverted(false);
+
+         left2 = new PWMVictorSPX(1);
+          addChild("left2",left2);
+          left2.setInverted(false);
+
+         SpeedControllerGroup leftMotor = new SpeedControllerGroup(left1, left2  );
+          addChild("Left Motor",leftMotor);
+
+
+         right1 = new PWMVictorSPX(5);
+          addChild("right1",right1);
+          right1.setInverted(false);
+
+         right2 = new PWMVictorSPX(6);
+          addChild("right2",right2);
+          right2.setInverted(false);
+
+         SpeedControllerGroup rightMotor = new SpeedControllerGroup(right1, right2  );
+          addChild("Right Motor",rightMotor);
+
+
+         drive = new DifferentialDrive(leftMotor, left1);
+          addChild("Drive",drive);
+          drive.setSafetyEnabled(true);
+         drive.setExpiration(0.1);
+         drive.setMaxOutput(1.0);
+
+             // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+             }
+
+             // Put methods for controlling this subsystem
+             // here. Call these from Commands.
+
+             public void drive(double left, double right) {
+                 drive.tankDrive(left, right);
+             }
+         }
```

Comment:
Should we run a formatter on this even though RobotBuilder does not?

Reply:
I think it would be confusing if the code in the documentation didn't match the generated code